packages/sdk/src/lib/authorize.ts

```diff
@@ -109,7 +109,21 @@ const authorize = async ({
             ]);
           }
 
-          const signature = splitSignature(parsedEAT.signature);
+          const { r, s } = secp256k1.Signature.fromCompact(
+            parsedEAT.signature.slice(2, 130)
+          );
+
+          const v = parseInt(`0x${parsedEAT.signature.slice(130)}`, 16) as
+            | 0
+            | 1
+            | 27
+            | 28;
+
+          const signature = {
+            r: `0x${r.toString(16)}`,
+            s: `0x${s.toString(16)}`,
+            v,
+          };
```

Comment:
Should we create a util function to parse the backend response / an EAT? This would avoid code duplication with `packages/sdk/src/hooks/useListenVioletEvents.tsx` AND we could export it for our partners to use.

Reply:
`npm install @violetprotocol/ethereum-access-token-helpers`

https://github.com/violetprotocol/ethereum-access-token/blob/main/src/utils/signature.ts#L7

😄

Reply:
Ah! Even better than an external package